src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/custom.py

```diff
@@ -0,0 +1,500 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+import os
+import shutil
+import adal  # pylint: disable=import-error
+from knack.prompting import prompt_y_n  # pylint: disable=unused-import
+from knack.util import CLIError
+from knack.log import get_logger
+from azure.cli.core._profile import Profile
+from azure.cli.command_modules.botservice._webutils import (
+    deploy_arm_template,
+    enable_zip_deploy,
+    get_app_settings,
+    _get_site_credential,
+    _get_scm_url)
+from azure.mgmt.botservice.models import Bot, BotProperties, sku
+
+logger = get_logger(__name__)
+
+
+def get_bot_site_name(endpoint):
+    import urllib.parse
+    split_parts = urllib.parse.urlsplit(endpoint)
+    return split_parts.netloc.split('.', 1)[0]
+
+
+def provisionConvergedApp(bot_name):
+    botfirstpartyid = 'f3723d34-6ff5-4ceb-a148-d99dcd2511fc'
+    aadclientid = '1950a258-227b-4e31-a9cf-717495945fc2'
+    tenantid = '72f988bf-86f1-41af-91ab-2d7cd011db47'
+
+    authority = 'https://login.windows.net/{0}'.format(tenantid)
+    context = adal.AuthenticationContext(
+        authority=authority,
+        validate_authority=True,
+        api_version=None
+    )
+
+    code = context.acquire_user_code(
+        resource=botfirstpartyid,
+        client_id=aadclientid,
+    )
+
+    logger.warning(code['message'])
+
+    token = context.acquire_token_with_device_code(
+        resource=botfirstpartyid,
+        user_code_info=code,
+        client_id=aadclientid
+    )
+    access_token = token['accessToken']
+
+    import requests
+    headers = {'Authorization': 'Bearer {0}'.format(access_token)}
+    response = requests.post(
+        'https://dev.botframework.com/api/botApp/provisionConvergedApp?name={0}'.format(bot_name),
+        headers=headers
+    )
+    if response.status_code not in [201]:
+        raise CLIError('Unable to provision appid and password for supplied credentials')
+    response_content = json.loads(response.content.decode('utf-8'))
+    msa_app_id = response_content['AppId']
+    password = response_content['Password']
+
+    return msa_app_id, password
+
+
+def create(cmd, client, resource_group_name, resource_name, kind, description=None, display_name=None,
+           endpoint=None, msa_app_id=None, password=None, tags=None, storageAccountName=None,
+           location='Central US', sku_name='F0', appInsightsLocation='South Central US',
+           language='Csharp', version='v3'):
+    if tags:
+        raise CLIError('Tags not supported yet')
```

Comment:
Aren't supported by the service or the command?  If they aren't supported by the service then you should just remove this. If they aren't supported by the command then you should support them in the command.

Reply:
removed the "not supported" line and have added support for tags